awx/ui/client/lib/components/syntax-highlight/syntax-highlight.directive.js

```diff
@@ -0,0 +1,89 @@
+const templateUrl = require('~components/syntax-highlight/syntax-highlight.partial.html');
+
+function atSyntaxHighlightController ($scope, AngularCodeMirror) {
+    const vm = this;
+    const varName = `${$scope.name}_codemirror`;
+
+    function init () {
+        if ($scope.disabled === 'true') {
+            $scope.disabled = true;
+        } else if ($scope.disabled === 'false') {
+            $scope.disabled = false;
+        }
+        $scope.value = $scope.value || $scope.default;
+
+        initCodeMirror();
+
+        $scope.$watch(varName, () => {
+            $scope.value = $scope[varName];
+        });
+    }
+
+    function initCodeMirror () {
+        $scope.varName = varName;
+        $scope[varName] = $scope.value;
+        const codeMirror = AngularCodeMirror(!!$scope.disabled);
+        codeMirror.addModes({
+            jinja2: {
+                mode: $scope.mode,
+                matchBrackets: true,
+                autoCloseBrackets: true,
+                styleActiveLine: true,
+                lineNumbers: true,
+                gutters: ['CodeMirror-lint-markers'],
+                lint: true,
+                scrollbarStyle: null,
+            }
+        });
+        codeMirror.showTextArea({
+            scope: $scope,
+            model: varName,
+            element: `${$scope.name}_codemirror`,
+            lineNumbers: true,
+            mode: $scope.mode,
+        });
+    }
+
+    vm.name = $scope.name;
+    vm.rows = $scope.rows || 6;
+    if ($scope.init) {
+        $scope.init = init;
+    }
+    angular.element(document).ready(() => {
+        init();
+    });
+    $scope.$on('reset-code-mirror', () => {
```

Comment:
I _think_ this approach will mean that every syntax highlight directive on the page will get re-rendered when the content of one of the others is updated.  In this case we probably don't care - I think there are only two of these on the page (?).  The only other approach that I can think of would be something like https://stackoverflow.com/questions/28116680/calling-directives-methods-from-parent-controller-in-angularjs#answer-33591133.  Honestly I'm not sure if it's worth changing but worth thinking about.

Reply:
Oh, maybe we only have one of these on the page (?).  I haven't actually spun it up and looked at the UI yet.

Reply:
Yeah, this message gets broadcast down from the form, so it refreshes all the syntax highlight inputs within the form -- which is what I want. It sends that request when it shows the inputs or changes which ones are visible